File: fixtures/scale/elixir/large/lib/support/bytes.ex
defmodule Support.Bytes do
  def zeros(n), do: List.duplicate(0, n)

  def add_all(k, bytes), do: Enum.map(bytes, &rem(&1 + k, 256))

  def rotate(n, bytes), do: Enum.drop(bytes, n) ++ Enum.take(bytes, n)
end

File: fixtures/scale/haskell/large/Support/Bytes.hs
module Support.Bytes (Bytes, zeros, addAll, rotate) where

type Bytes = [Int]

zeros :: Int -> Bytes
zeros n = replicate n 0

addAll :: Int -> Bytes -> Bytes
addAll k = map (\b -> mod (b + k) 256)

rotate :: Int -> Bytes -> Bytes
rotate n xs = drop n xs ++ take n xs
